avalon/tools/lib.py

```diff
@@ -107,6 +107,75 @@ def iter_model_rows(model,
         yield index
 
 
+@contextlib.contextmanager
+def preserve_states(
```

Comment:
This formatting, I see it a lot lately. Is this pycharm or black or something? Let's stick with having the first argument on the same line as usual, and every other argument on a new line with the same indentation.

Reply:
> Is this pycharm or black or something?

This is actually Me :) . I personally don't like that strange offset when first argument is on the same line. It's hard to read and edit for me and in this case it would cause PEP 8 E501. But I agree there should be one argument per line.